roles/lib_openshift/library/oc_obj.py

```diff
@@ -1533,14 +1533,11 @@ def run_ansible(params, check_mode=False):
         if state == 'list':
             return {'changed': False, 'results': api_rval, 'state': 'list'}
 
-        if not params['name']:
-            return {'failed': True, 'msg': 'Please specify a name when state is absent|present.'}  # noqa: E501
-
         ########
         # Delete
         ########
         if state == 'absent':
-            if not Utils.exists(api_rval['results'], params['name']):
+            if not api_rval['results'] or not api_rval['results'][0]:
```

Comment:
If ``api_rval['results']`` could be a true statement and also not a list then this could cause errors.

If that's the case, maybe something like:

```python
if not api_rval['results'] or (isinstance(api_rval['results'], list) and not api_rval['results'][0]):
```

Otherwise it's fine. Just double checking.